zc_common/remote_resource/tests.py

```diff
@@ -12,11 +12,54 @@ class ResponseTestCase(APITestCase):
     FAILURE_HIGH_LEVEL_KEYS = ['errors']
     FAILURE_DATA_KEYS = ['status', 'source', 'detail']
 
+    INSTANCE_TOP_LEVEL_KEYS = ["type", "id", "attributes"]
+
     def convert_to_list(self, data):
-        if isinstance(data, list):
+        if isinstance(data, dict):
+            return [data]
+        elif isinstance(data, list):
             return data
         else:
-            return [data]
+            return data.all()
+
+    def validate_instance_in_data(self, data, instance, attributes, relationship_keys=None):
+        self.assertTrue(all(key in data for key in self.INSTANCE_TOP_LEVEL_KEYS))
+
+        instance_attributes = data["attributes"]
+        for key in attributes:
+            self.assertEqual(instance_attributes[camelize(key, False)], getattr(instance, key))
+
+        if not relationship_keys:
+            return True
```

Comment:
I think the timing of the return will be unaffected by changing the if statement. In both cases it has to evaluate whether `relationship_keys` is None or not, and in the way I've suggested it'll evaluate to None, skip the entire rest of the function, and return.

My personal opinion is that it would be cleaner to change it as I've suggested. It would remove unnecessary lines and I don't think the readability would be impacted too much by an additional tab. I'm generally against returning dummy values that are ignored (if you want to keep it, I'd at least change it to a bare `return`).

That all said, if you feel strongly about keeping it as is I suppose I'm ok with that. It's not unclear as written, but my preference would still be for changing it.